.github/workflows/continuous-integration.yml

```diff
@@ -6,71 +6,73 @@ on:
 
 name: "Continuous Integration"
 
+env:
+  PHP_EXTENSIONS: "json, zip, opcache, mbstring"
+  PHP_INI_VALUES: "opcache.enable=1, opcache.enable_cli=1, opcache.fast_shutdown=0, zend.assertions=1, assert.exception=On"
+
 jobs:
   validate:
-    name: "Validate"
+    name: "Validate composer"
 
-    runs-on: "ubuntu-latest"
+    runs-on: "${{ matrix.operating-system }}"
+
+    continue-on-error: false
 
     strategy:
       fail-fast: false
+
       matrix:
-        php-versions:
-          - "7.2"
+        operating-system:
+          - "ubuntu-20.04"
+
+        php-version:
+          - "7.4"
 
         dependencies:
           - ""
 
     steps:
       - name: "Checkout"
         uses: "actions/checkout@v2.4.0"
+        with:
+          # Disabling shallow clone is recommended for improving relevancy of reporting
+          fetch-depth: 0
 
       - name: "Install PHP"
-        uses: "shivammathur/setup-php@2.16.0"
+        uses: "shivammathur/setup-php@2.15.0"
         with:
-          php-version: "${{ matrix.php-versions }}"
-          extensions: "json, zip" #optional
-          ini-values: "opcache.enable=1, opcache.enable_cli=1, zend.assertions=1, assert.exception=On" #optional
-          coverage: "none" #optional
+          php-version: "${{ matrix.php-version }}"
+          extensions: "${{ env.PHP_EXTENSIONS }}"
+          ini-values: "${{ env.PHP_INI_VALUES }}"
+          coverage: "none"
+          tools: "composer:v2, composer-normalize"
 
       - name: "Validate composer.json"
         run: "composer validate --strict"
 
-      - name: "Determine composer cache directory"
-        id: "determine-composer-cache-directory"
-        run: "echo \"::set-output name=directory::$(composer config cache-dir)\""
-
-      - name: "Cache dependencies installed with composer"
-        uses: "actions/cache@v2.1.7"
-        with:
-          path: "${{ steps.determine-composer-cache-directory.outputs.directory }}"
-          key: "php-${{ matrix.php-versions }}-composer-${{ matrix.dependencies }}-${{ hashFiles('composer.lock') }}"
-          restore-keys: "php-${{ matrix.php-versions }}-composer-${{ matrix.dependencies }}-"
-
-      - name: "Update dependencies with composer"
-        run: "composer update --optimize-autoloader --prefer-dist --prefer-stable --no-progress --no-interaction -vv --ansi ${{ matrix.dependencies }}"
-
       - name: "Run composer-normalize"
-        uses: "docker://ergebnis/composer-normalize-action:latest"
-        with:
-          args: "--dry-run"
+        run: "composer-normalize --dry-run"
 
   install:
-    name: "Check dependencies"
+    name: "Install dependencies"
 
     needs: "validate"
 
-    runs-on: "ubuntu-latest"
+    runs-on: "${{ matrix.operating-system }}"
+
+    continue-on-error: false
 
     strategy:
       fail-fast: false
+
       matrix:
-        php-versions:
-          - "7.1"
-          - "7.2"
-          - "7.3"
+        operating-system:
+          - "ubuntu-18.04"
```

Comment:
Do we need to test on such an old `ubuntu` still? 20.04 is the latest LTS and is sufficient IMO

Reply:
In the company I am working we are using this version, and there are currently no plans to update. To make sure, I am able to install the project here to work on it sometimes, I would like to keep the old ubuntu version for this test.